### PR TITLE
Sync -chef/go-chef into go-chef/chef as of 07/08/2020

### PR DESCRIPTION
**Pull request by marcparadise**
Brings chef/go-chef up-to-date with go-chef/chef as of 8 Jul, 2020.

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
-This Library can be used to write tools to interact with the chef server.
+This is a Chef Infra Server API client. This Library can be used to write tools to
+interact with the chef server.
 
-The testing can be run with `go test`, and the client can be used as per normal via `go get github.com/go-chef/chef`
-Documentation can be found on GoDoc at  http://godoc.org/github.com/go-chef/chef
+The testing can be run with go test, and the client can be used as per normal via:
 
-This is example code generating a new node on the chef-server.
+		go get github.com/chef/go-chef
```
Comment (MarkGibbons) [inline doc.go:7]: different here.

File: test_chef_server/kitchen.yml [hunk 2]
```diff
@@ -21,7 +25,10 @@ suites:
       - recipe[test_chef_server::default]
       - recipe[test_chef_server::chef_objects]
       - recipe[test_chef_server::chefapi]
+<<<<<<< HEAD
       - recipe[test_chef_server::workstation]
+=======
+>>>>>>> master
```
Comment (tas50) [inline test_chef_server/kitchen.yml:31]: There's some merge artifacts

**Comment by MarkGibbons** [comment]
I think I've merged a lot of these changes already.  Let me look things over. Most of the PR looks fairly simple to beat into shape by renaming files.  The merge artifacts left over will be a bit painful.

File: test_chef_server/kitchen.yml (hunk 2 above)
Comment (marcparadise) [reply test_chef_server/kitchen.yml:31]: Ugh yeah, all over the place in `examples` and `testapi`

**Comment by marcparadise** [comment]
Okay, thanks @MarkGibbons .It starts to look we didn't have anything in `examples` or `testapi`, and the conflicts are from a previous merge.

**Comment by marcparadise** [comment]
After confirming that we had no changes in `examples` downstream that were not already captured here post repo-reorg, I was able to remove the `examples` directory from this PR, and restore the symlink. Git wasn't able to recognize them as the same objects due to one (or more?) prior merges of content from chef/go-chef into those paths. 

This has made the diff relatively sane.

**Comment by marcparadise** [comment]
It looks like this should get us there. I'm fix the .mod/.sum/doc.go files and take a final pass to make sure no changes from the chef-fork side are lost in the mass cleanup of `testapi` and `examples`.

**Comment by MarkGibbons** [comment]
I'll take a look, today or tomorrow.  Thanks very much for this one.

**Comment by MarkGibbons** [comment]
I found a few minor things that I'll take care of when I merge it.  What I'd like to see is integration tests, I should be able to add those too.

**Comment by marcparadise** [comment]
Let me also verify that I didn't erase any that we added, that might have been hidden by the renames.

**Comment by marcparadise** [comment]
doc.go has been reverted to this repo's version - none of our changes should be pulled in for that file. 

I've confirmed that the commits here don't contain any additions/modifications to the integration tests from our side since last sync-up.

**Comment by marcparadise** [comment]
I've updated go.mod as well post-merge. I think that puts this PR in its final state in terms of correctness of merge.

**Comment by MarkGibbons** [comment]
Thanks, I'll get to it today.

**Pull request merged**